### PR TITLE
Add timeout filter action

### PR DESCRIPTION
**Pull request by SilverAndro**
Think i did this right, dont really have time to test :shipit:

### CODE REVIEW

**Review by sschr15 — APPROVED** [review]
Great, but future work might want to add a method by which to vary the timeout length

**Comment by SilverAndro** [comment]
10 minutes was chosen more as a pause on a situation until a mod has time to review whats going on than a punishment on a user.

File: src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/filtering/FilterAction.kt [hunk 1]
```diff
@@ -17,6 +17,7 @@ enum class FilterAction(
 ) : ChoiceEnum {
 	RESPOND(-1, "Respond with the filter note"),
 	DELETE(0, "Delete message"),
-	KICK(1, "Kick user", true),
-	BAN(2, "Ban user", true)
+	TIMEOUT(1, "Timeout user", true),
+	KICK(2, "Kick user"),
+	BAN(3, "Ban user", true)
```
Comment (gdude2002) [inline src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/filtering/FilterAction.kt:22]: Won't this break all existing filter entries?

Comment (SilverAndro) [reply src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/filtering/FilterAction.kt:22]: I only saw severity being used for sorting, but no clue how it happens internally

Comment (gdude2002) [reply src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/filtering/FilterAction.kt:22]: I'm pretty sure KMongo is using the number for database storage, but I'd have to check

Comment (gdude2002) [reply src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/filtering/FilterAction.kt:22]: Ah, no, it's not, this is OK then

image

**Comment by gdude2002** [comment]
Good stuff!

**Pull request merged**